include/posix/mqueue.h

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __MQUEUE_H__
+#define __MQUEUE_H__
+
+#include <kernel.h>
+#include <time.h>
+#include "sys/types.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef void *mqd_t;
+typedef unsigned int mode_t;
+
+typedef struct mq_attr {
+	long mq_flags;
+	long mq_maxmsg;
+	long mq_msgsize;
+	long mq_curmsgs;	/* Number of messages currently queued. */
+} mq_attr;
+
+#define O_CREAT_POS	9
+#define O_CREAT         (1 << O_CREAT_POS)
+
+#define O_EXCL_POS	11
+#define O_EXCL          (1 << O_EXCL_POS)
+
+#define O_NONBLOCK_POS	14
+#define O_NONBLOCK      (1 << O_NONBLOCK_POS)
+
+#define O_RDONLY        0
+#define O_WRONLY        1
+#define O_RDWR          2
```

Comment:
Do we really not have these defined somewhere else already?  Seems like something else would have duplicated them, which means this header can collide.  Properly they should be going into a fcntl.h header, right?

Reply:
yes it is already part of <sys/functl.h> which is part of tool chain. yes we can include this file.

Reply:
I am getting crazy compilation issues if i include <fcntl.h>. 
as part of fcntl.h following 2 files are getting loaded - 
#include <sys/types.h>
#include <sys/stat.h>           /* sigh. for the mode bits for open/creat */

inside stat.h it complains multiple variables (like dev_t etc) are undefined. While looking at sys/types.h, i get a note that 
/*
 * All these should be machine specific - right now they are all broken.
 * However, for all of Cygnus' embedded targets, we want them to all be
 * the same.  Otherwise things like sizeof (struct stat) might depend on
 * how the file was compiled (e.g. -mint16 vs -mint32, etc.).
 */

#ifndef __CYGWIN__      /* which defines these types in it's own types.h. */
typedef _off_t  off_t;
typedef __dev_t dev_t;
typedef __uid_t uid_t;
typedef __gid_t gid_t;
#endif

As of now I have put a FIXME note in POSIX layer, I will revisit this later.
By that time can you please have a look at the new patch set.

Reply:
Can you at least protect it?  Right now if you have an app that tries to include those other definitions and POSIX, it will fail with a preprocessor redefinition error.